app/services/invitation_handler.rb

```diff
@@ -0,0 +1,19 @@
+class InvitationHandler
+  
+  def initialize(options={})
+    @user = options[:user]
+    @token = options[:invitation_token]
+  end
+  
+  def handle_invitation
+    invitation = Invitation.find_by_token(token)
```

Comment:
I would prefer `.find_by(token: token)` over `.find_by_token(token)` for 2 reasons: 1) find_by_xxx is using method_missing and it is not very efficient. 2) I don't find it much readable - but this is more personal opinion.